### PR TITLE
Feature/outlined input

### LINKED ISSUE #1752: Outlined Inputs
**Enhancement**

Add support for outlined inputs as per the material.io guidelines https://material.io/components/text-fields#input-types.

This should use a property `kind` to determine the type of input and supported by all the different types of input such as email-input, password-input, select etc

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** bug / feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] For new widgets, `theme.variant()` is added to the root domnode
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:**
Add outlined option "kind" to `text-input` and `text-area` widgets

**Text Input**
text-input

**Text Area**
text-area

Resolves #1752 
Supersedes: #1756

### CODE REVIEW

File: src/chip-typeahead/index.tsx [hunk 1]
```diff
@@ -42,6 +40,8 @@ export interface ChipTypeaheadProperties {
 	duplicates?: boolean;
 	/** Flag to indicate if values other than those in the resource can be entered, defaults to true */
 	strict?: boolean;
+	/** The kind of chip typeahead input */
+	kind?: 'outlined' | 'default';
```
Comment (agubler) [inline src/chip-typeahead/index.tsx:44]: Please update `default` to `filled`

File: src/chip-typeahead/index.tsx [hunk 2]
```diff
@@ -255,10 +234,17 @@ export const ChipTypeahead = factory(function ChipTypeahead({
...
-						inputWrapper: [themeCss.inputWrapper],
 						input: [themeCss.input],
 						wrapper: [themeCss.wrapper],
-						leading: [themeCss.inputLeading]
+						leading: [themeCss.inputLeading],
+						leadingWrapper: [themeCss.inputLeadingWrapper],
+						defaultKind: [themeCss.defaultKind],
+						outlinedKind: [themeCss.outlinedKind],
+						labelWrapper: [themeCss.inputLabel],
+						label: [themeCss.label, chips.length ? themeCss.labelActive : undefined],
```
Comment (agubler) [inline src/chip-typeahead/index.tsx:244]: `chips.length && themeCss.labelActive`

File: src/chip-typeahead/index.tsx (hunk 1 above)
Comment (tomdye) [reply src/chip-typeahead/index.tsx:44]: good idea

File: src/chip-typeahead/index.tsx (hunk 2 above)
Comment (tomdye) [reply src/chip-typeahead/index.tsx:244]: not sure it liked that when i tried before

File: src/text-input/index.tsx [hunk 3]
```diff
@@ -240,95 +286,86 @@ export const TextInput = factory(function TextInput({
...
-				)}
-				<div
-					key="inputWrapper"
-					classes={[
-						themeCss.inputWrapper,
-						inputFocused ? themeCss.inputWrapperFocused : undefined
-					]}
-					role="presentation"
-				>
+				{_renderLabel()}
```
Comment (agubler) [inline src/text-input/index.tsx:296]: underscore prefixed function?

File: src/theme/material/variants/dark.m.css [hunk 4]
```diff
@@ -1,4 +1,6 @@
-.root {
+.root,
+.root *::before,
```
Comment (agubler) [inline src/theme/material/variants/dark.m.css:2]: @tomdye what's this about?

**Review by agubler — APPROVED** [review]
A couple of nits but otherwise looks good, nice work @KaneFreeman / @tomdye

File: src/theme/material/variants/dark.m.css (hunk 4 above)
Comment (KaneFreeman) [reply src/theme/material/variants/dark.m.css:2]: This is necessary to pass the css variables down to the before and after selectors

Comment (tomdye) [reply src/theme/material/variants/dark.m.css:2]: Why do the before / after nodes need them?

**Pull request merged**